pkg/network/protocols/kafka/api_version_counter.go

```diff
@@ -0,0 +1,63 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package kafka
+
+import (
+	libtelemetry "github.com/DataDog/datadog-agent/pkg/network/protocols/telemetry"
+)
+
+// APIVersionCounter is a Kafka API version aware counter, it has a counter for each supported Kafka API version.
+// It enables the use of a single metric that increments based on the API version, avoiding the need for separate metrics for API version
+type APIVersionCounter struct {
+	hitsV1, hitsV2, hitsV3, hitsV4, hitsV5, hitsV6, hitsV7, hitsV8, hitsV9, hitsV10, hitsV11 *libtelemetry.Counter
+}
```

Comment:
consider using [prometheus.CounterVec](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#CounterVec) instead

[example](https://github.com/DataDog/datadog-agent/blob/2b792e68d8f2f9967d01b278e3b764118d5933ca/pkg/ebpf/telemetry/perf_metrics.go#L29) from our code

P.S: there are also some wrappers ([1](https://github.com/DataDog/datadog-agent/blob/main/comp/core/telemetry/prom_counter.go), [2](https://github.com/DataDog/datadog-agent/blob/main/pkg/telemetry/counter.go)) that we have in our repo

Reply:
This is only relevant for prometheus and not for statsd metrics